### PR TITLE
[core][branch] Add branches in tags table

### LINKED ISSUE #2753: [Feature] Add branches in tags table
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.

### Motivation

Add branches column in tags table, subtask of #1795 

### Solution

_No response_

### Anything else?

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by FangYongs**
### Purpose

Linked issue: close #2753 

Add branches column in tags table

### Tests

Added test in `TagsTableTest` to get branches

### API and Format

no

### Documentation

yes

### CODE REVIEW

File: docs/content/how-to/system-tables.md [hunk 1]
```diff
@@ -198,12 +198,12 @@ and some historical information of the snapshots. You can also get all tag names
...
-+----------+-------------+-----------+-------------------------+--------------+
-| tag_name | snapshot_id | schema_id |             commit_time | record_count |
-+----------+-------------+-----------+-------------------------+--------------+
-|     tag1 |           1 |         0 | 2023-06-28 14:55:29.344 |            3 |
-|     tag3 |           3 |         0 | 2023-06-28 14:58:24.691 |            7 |
-+----------+-------------+-----------+-------------------------+--------------+
++----------+-------------+-----------+-------------------------+--------------+--------------+
+| tag_name | snapshot_id | schema_id |             commit_time | record_count |   branches   |
++----------+-------------+-----------+-------------------------+--------------+--------------+
+|     tag1 |           1 |         0 | 2023-06-28 14:55:29.344 |            3 |      []      |
```
Comment (TaoZex) [inline docs/content/how-to/system-tables.md:204]:
```suggestion
|     tag1 |           1 |         0 | 2023-06-28 14:55:29.344 |            3 |      [main]      |
```
How about use `main` ?

File: paimon-core/src/main/java/org/apache/paimon/table/system/TagsTable.java [hunk 2]
```diff
@@ -202,15 +221,18 @@ public RecordReader<InternalRow> createReader(Split split) {
...
+            List<String> branches = tagBranches.get(tag.getValue());
             return GenericRow.of(
                     BinaryString.fromString(tag.getValue()),
                     snapshot.id(),
                     snapshot.schemaId(),
                     Timestamp.fromLocalDateTime(
                             DateTimeUtils.toLocalDateTime(snapshot.timeMillis())),
-                    snapshot.totalRecordCount());
+                    snapshot.totalRecordCount(),
+                    BinaryString.fromString(branches == null ? "[]" : branches.toString()));
```
Comment (TaoZex) [inline paimon-core/src/main/java/org/apache/paimon/table/system/TagsTable.java:235]: same as above

**Comment by FangYongs** [comment]
This PR will wait for #2748

File: paimon-core/src/main/java/org/apache/paimon/utils/TagManager.java [hunk 3]
```diff
@@ -72,6 +72,26 @@ public Path branchTagPath(String branchName, String tagName) {
...
+                    listVersionedFileStatus(
+                                    fileIO,
+                                    new Path(getBranchPath(tablePath, branchName) + "/tag/"),
+                                    TAG_PREFIX)
+                            .map(FileStatus::getPath)
+                            .collect(Collectors.toList());
+            checkArgument(
+                    tagPaths.size() == 1,
+                    String.format(
+                            "There are multiple tags %s for branch %s",
```
Comment (schnappi17) [inline paimon-core/src/main/java/org/apache/paimon/utils/TagManager.java:87]: I'm confused here, why it's not allowed to generate more tags on branch?

File: docs/content/how-to/system-tables.md (hunk 1 above)
Comment (FangYongs) [reply docs/content/how-to/system-tables.md:204]: I think it's better to given empty branches instead of `main` if there is no branch for the tag

**Comment by schnappi17** [comment]
+1

**Pull request merged**